src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -353,6 +355,91 @@ public String addDatasetAssetSpecifications(
         .submit();
   }
 
+  public void validateDatasetAssetSpecification(Dataset dataset, AssetModel assetModel) {
+    // Validate Root Table
+    DatasetTable rootTable =
+        dataset.getTables().stream()
+            .filter(datasetTable -> datasetTable.getName().equals(assetModel.getRootTable()))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new InvalidAssetException(
+                        "Root table " + assetModel.getRootTable() + " does not exist in dataset."));
+    // Validate Root Column
+    if (!rootTable.getColumns().stream()
+        .map(c -> c.getName())
+        .toList()
+        .contains(assetModel.getRootColumn())) {
```

Comment:
In general it's not good to `toList()` on a stream if you have more operations to perform on its elements. Using `toList()` forces the entire stream into memory, whereas using another streaming operation means that only one object needs to be in memory at a time. I think `anyMatch(c -> c.getName().equals(assetModel.getRootColumn())` would work here.